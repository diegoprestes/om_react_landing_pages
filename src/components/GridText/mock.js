export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus rutrum est a ex hendrerit malesuada. Aenean ornare placerat sapien, in lobortis odio suscipit et. Etiam eu erat elit. Nulla nec magna orci. Duis sit amet lectus a metus commodo ullamcorper nec nec nulla. Ut laoreet blandit arcu ac finibus. Proin finibus, risus porttitor sagittis mattis, orci mauris vehicula odio, quis suscipit erat diam a ante. Nullam vel metus ipsum.'
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus rutrum est a ex hendrerit malesuada. Aenean ornare placerat sapien, in lobortis odio suscipit et. Etiam eu erat elit. Nulla nec magna orci. Duis sit amet lectus a metus commodo ullamcorper nec nec nulla. Ut laoreet blandit arcu ac finibus. Proin finibus, risus porttitor sagittis mattis, orci mauris vehicula odio, quis suscipit erat diam a ante. Nullam vel metus ipsum.'
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus rutrum est a ex hendrerit malesuada. Aenean ornare placerat sapien, in lobortis odio suscipit et. Etiam eu erat elit. Nulla nec magna orci. Duis sit amet lectus a metus commodo ullamcorper nec nec nulla. Ut laoreet blandit arcu ac finibus. Proin finibus, risus porttitor sagittis mattis, orci mauris vehicula odio, quis suscipit erat diam a ante. Nullam vel metus ipsum.'
    }
  ]
};
