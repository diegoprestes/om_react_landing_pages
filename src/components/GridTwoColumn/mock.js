export default {
  title: 'Grid two columns',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Cum esse eveniet nam nisi repudiandae, explicabo non minus perspiciatis? Laudantium et dolorem omnis asperiores! Suscipit unde id dolore, provident ab quibusdam?`,
  srcImg: 'assets/images/javascript.svg',
  hasBackground: false
};
