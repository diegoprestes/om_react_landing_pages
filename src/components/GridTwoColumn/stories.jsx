import GridTwoColumn from '.';

import args from './mock';

export default {
  title: 'GridTwoColumn',
  component: GridTwoColumn,
  args: args
};

export const Template = (args) => {
  return (
    <div>
      <GridTwoColumn {...args} />
    </div>
  );
};
