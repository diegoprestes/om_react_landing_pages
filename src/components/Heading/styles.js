import styled, { css } from 'styled-components';

const mediaFont = (theme) => css`
  @media ${theme.media.sm} {
    font-size: ${theme.fonts.sizes['4xl']};
  }
`;

const titleSize = {
  sm: (theme) => css`
    font-size: ${theme.fonts.sizes.md};
  `,
  md: (theme) => css`
    font-size: ${theme.fonts.sizes.lg};
  `,
  lg: (theme) => css`
    font-size: ${theme.fonts.sizes.xl};
  `,
  xl: (theme) => css`
    font-size: ${theme.fonts.sizes.lg};
    ${mediaFont(theme)};
  `
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)}
  `}
`;
