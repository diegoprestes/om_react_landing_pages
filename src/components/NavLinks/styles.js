import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column nowrap;
    align-items: center;

    @media ${theme.media.sm} {
      flex-flow: row wrap;
    }
  `}
`;
