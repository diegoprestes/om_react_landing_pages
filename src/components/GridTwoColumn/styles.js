import styled, { css } from 'styled-components';

import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr;
    align-items: center;
    gap: ${theme.spacings.lg};

    @media ${theme.media.sm} {
      grid-template-columns: 1fr 2fr;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xl};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.lg};

    @media ${theme.media.sm} {
      margin-bottom: 0;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css``}
`;
