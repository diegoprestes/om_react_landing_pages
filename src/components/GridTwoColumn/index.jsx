import PropTypes from 'prop-types';
import * as Styled from './styles';

import Heading from '../Heading';
import SectionBackground from '../SectionBackground';
import TextComponent from '../TextComponent';

const GridTwoColumn = ({ title, text, srcImg, hasBackground = false }) => {
  return (
    <SectionBackground hasBackground={hasBackground}>
      <Styled.Container>
        <Styled.TextContainer>
          <Heading uppercase colorDark={!hasBackground} as="h2">
            {title}
          </Heading>
          <TextComponent>{text}</TextComponent>
        </Styled.TextContainer>
        <Styled.ImageContainer>
          <Styled.Image src={srcImg} alt={title} />
        </Styled.ImageContainer>
      </Styled.Container>
    </SectionBackground>
  );
};

GridTwoColumn.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  srcImg: PropTypes.string.isRequired,
  hasBackground: PropTypes.bool
};

export default GridTwoColumn;
