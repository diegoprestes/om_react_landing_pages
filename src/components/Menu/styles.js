import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Container as SectionBackground } from '../SectionBackground/styles';
import { Title as Heading } from '../Heading/styles';

const menuVisible = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 5;
    background: ${theme.colors.white};
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    transition: all 300ms ease-in-out;
    ${visible && menuVisible(theme)}

    > ${SectionContainer} {
      display: grid;
      grid-template-columns: 1fr;
      grid-template-rows: 1fr;
      height: 100vh;
      overflow-y: auto;
      align-items: center;
      padding: 0 ${theme.spacings.lg};
    }

    > ${SectionBackground} {
      min-height: 100vh;
      display: flex;
      align-items: center;
    }

    & ${Heading} {
      display: flex;
      justify-content: center;
      overflow-y: none;
      padding-bottom: ${theme.spacings.lg};
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.sm} {
      position: static;
      height: auto;
      border-bottom: 1px solid ${theme.colors.mediumGray};
      ${menuVisible(theme)};

      > ${SectionContainer} {
        display: block;
        height: auto;
        padding: 0;
      }

      & ${Heading} {
        display: block;
        padding: 0;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: ${theme.spacings.xl} 0;

    @media ${theme.media.sm} {
      display: flex;
      justify-content: space-between;
      align-items: center;
      padding: 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    display: block;
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    cursor: pointer;

    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }

    @media ${theme.media.sm} {
      display: none;
    }
  `}
`;
