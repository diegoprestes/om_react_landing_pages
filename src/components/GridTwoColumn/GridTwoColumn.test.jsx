import GridTwoColumn from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<GridTwoColumn />', () => {
  it('should render grid with two columns', () => {
    const { container } = renderTheme(<GridTwoColumn {...mock} />);

    expect(container).toMatchSnapshot();
  });
});
