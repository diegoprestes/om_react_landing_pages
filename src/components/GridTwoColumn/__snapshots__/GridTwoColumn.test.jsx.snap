// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render grid with two columns 1`] = `
.c5 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

.c3 {
  margin-bottom: 3.2rem;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
}

.c6 {
  font-size: 2.4rem;
}

@media (min-width:768px) {
  .c5 {
    font-size: 6.4rem;
  }
}

@media (min-width:768px) {
  .c2 {
    grid-template-columns: 1fr 2fr;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-bottom: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h1
            class="c4 c5"
          >
            Grid two columns
          </h1>
          <p
            class="c6"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Cum esse eveniet nam nisi repudiandae, explicabo non minus perspiciatis? Laudantium et dolorem omnis asperiores! Suscipit unde id dolore, provident ab quibusdam?
          </p>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid two columns"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
